raiden_contracts/contracts/TokenNetwork.sol

```diff
@@ -243,22 +244,27 @@ contract TokenNetwork is Utils {
 
     /// @notice Sets the channel participant total deposit value.
     /// Can be called by anyone.
+    /// @param channel_identifier Identifier for the channel on which this operation takes place.
     /// @param participant Channel participant whose deposit is being set.
     /// @param total_deposit The total amount of tokens that the participant will have
     /// as a deposit.
     /// @param partner Channel partner address, needed to compute the channel identifier.
-    function setTotalDeposit(address participant, uint256 total_deposit, address partner)
-        isOpen(participant, partner)
+    function setTotalDeposit(
+        uint256 channel_identifier,
+        address participant,
+        uint256 total_deposit,
+        address partner
+    )
+        isOpen(channel_identifier)
         public
     {
+        require(channel_identifier == getChannelIdentifier(participant, partner));
```

Comment:
okay, so if I understood correctly we cannot have just the channel identifier because we need to add the deposit of both participants and the addresses of each participant are not available, correct?

Reply:
Correct.